custom/src/main/java/com/splunk/opentelemetry/appd/AppdBonusConfigurationCustomizerProvider.java

```diff
@@ -0,0 +1,100 @@
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.PropagatorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.TracerProviderModel;
+import java.util.Map;
+
+@AutoService(DeclarativeConfigurationCustomizerProvider.class)
+public final class AppdBonusConfigurationCustomizerProvider
+    implements DeclarativeConfigurationCustomizerProvider {
+
+  private static final String CONFIG_CISCO_CTX_ENABLED = "cisco.ctx.enabled";
+
+  @Override
+  public void customize(DeclarativeConfigurationCustomizer autoConfiguration) {
+    autoConfiguration.addModelCustomizer(
+        model -> {
+          if (model.getInstrumentationDevelopment() == null
+              || model.getInstrumentationDevelopment().getJava() == null) {
+            return model;
+          }
+          Map<String, Object> properties =
+              model.getInstrumentationDevelopment().getJava().getAdditionalProperties();
+
+          if (isFeatureEnabled(model, properties)) {
+            if (maybeAddAppdBonusPropagator(model)) {
+              // Appd propagator has been added so add also a corresponding Appd span processor
+              SpanProcessorModel appdSpanProcessorModel =
+                  new SpanProcessorModel()
+                      .withAdditionalProperty(AppdBonusSpanProcessorComponentProvider.NAME, null);
+              if (model.getTracerProvider() == null) {
+                model.withTracerProvider(new TracerProviderModel());
+              }
+              model.getTracerProvider().getProcessors().add(appdSpanProcessorModel);
+            }
+          }
+
+          return model;
+        });
+  }
+
+  private static boolean isFeatureEnabled(
+      OpenTelemetryConfigurationModel model, Map<String, Object> properties) {
+    return (getAdditionalPropertyOrDefault(properties, CONFIG_CISCO_CTX_ENABLED, false));
+  }
+
+  private static boolean canAddPropagator(String compositeList) {
+    for (String propagatorNames : compositeList.split(",")) {
+      if (propagatorNames.trim().equals("none")) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private static boolean maybeAddAppdBonusPropagator(OpenTelemetryConfigurationModel model) {
+    if (model.getPropagator() == null) {
+      model.withPropagator(new PropagatorModel());
+    }
+
+    String compositeList = model.getPropagator().getCompositeList();
+    if (compositeList == null) {
+      compositeList = "";
+    } else if (!canAddPropagator(compositeList)) {
+      return false;
+    }
+
+    // Possible duplicates are handled by the upstream
+    compositeList =
+        compositeList.isEmpty()
+            ? AppdBonusPropagator.NAME
+            : AppdBonusPropagator.NAME + "," + compositeList;
```

Comment:
Without declarative config we need to add the default propagators. Just checking whether this is also needed here or is it so that when you don't add propagators to yaml then you don't get any default ones?